src/iov.c

```diff
@@ -86,3 +86,18 @@ void ofi_consume_iov(struct iovec *iov, size_t *iov_count, size_t consumed)
 	iov[0].iov_base = (uint8_t *)iov[0].iov_base + consumed;
 	iov[0].iov_len -= consumed;
 }
+
+int ofi_truncate_iov(struct iovec *iov, size_t *iov_count, size_t trim_size)
+{
+	size_t i = 0;
+
+	for (i = 0; i < *iov_count; i++) {
+		if (trim_size <= iov[i].iov_len) {
+			iov[i].iov_len = trim_size;
+			*iov_count = i + 1;
+			return FI_SUCCESS;
+		}
+		trim_size -= iov[i].iov_len;
+	}
+	return -FI_EINVAL;
+}
```

Comment:
The calling code doesn't use the return value, so we can convert this function to void.